src/server/req_delete.c

```diff
@@ -236,30 +236,68 @@ check_deletehistoryjob(struct batch_request * preq)
 		}
 	}
 
-	historyjob = svr_chk_histjob(histpjob);
-	if (historyjob == PBSE_HISTJOBID) {
+	if (svr_chk_histjob(pjob) != PBSE_HISTJOBID) {
+		/*
+		 *  Job is in a Non Finished state . It must be terminated and then its history
+		 * should be purged .
+		 */
+		return FALSE;
+	}
+
+	/* Issue history job delete request to remote server if job is moved. */
+	if (pjob->ji_qs.ji_state == JOB_STATE_MOVED)
+		issue_delete(pjob);
+
+	if (jt == IS_ARRAY_NO) {
 		snprintf(log_buffer, sizeof(log_buffer),
 			msg_job_history_delete, preq->rq_user,
 			preq->rq_host);
 		log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_JOB, LOG_INFO,
 			jid,
 			log_buffer);
 
-		/* Issue history job delete request to remote server if job is moved. */
-		if (histpjob->ji_qs.ji_state == JOB_STATE_MOVED)
-			issue_delete(histpjob);
+		job_purge(pjob);
+		preq->rq_reply.brp_code = PBSE_HISTJOBDELETED;
+		reply_send(preq);
+		return TRUE;
+	} else if (jt == IS_ARRAY_ArrayJob) {
```

Comment:
Since history of subjobs is also getting deleted now we need to announce that in EDD as this is behavior change as discussed.

Reply:
Reverted in rc 20 https://github.com/PBSPro/pbspro/pull/590/commits/b01fcac3e94363966f635e6fad726a5342288a79